File: apps/admin-v2/src/infrastructure/api/repositories/api-auth.repository.ts
import type {
  AuthRepository,
  LoginResult,
  RegisterResult,
} from '@/domain/repositories/auth.repository';
import api from '../client';
import { mapUser } from '../mappers/user.mapper';
import { mapTenant } from '../mappers/tenant.mapper';

export class ApiAuthRepository implements AuthRepository {
  async login(identifier: string, password: string): Promise<LoginResult> {
    const trimmed = identifier.trim();
    const isEmail = trimmed.includes('@');
    const payload = isEmail
      ? { email: trimmed, password }
      : { identifier: trimmed, password };
    const { data: res } = await api.post('/auth/login', payload);
    const d = res.data;
    return {
      user: mapUser(d.user),
      token: d.token,
      tenant: d.tenant ? mapTenant(d.tenant) : null,
    };
  }

  async register(params: {
    name: string;
    email: string;
    password: string;
    businessName?: string;
    businessType?: string;
  }): Promise<RegisterResult> {
    const { data: res } = await api.post('/auth/register', {
      name: params.name,
      email: params.email,
      password: params.password,
      business_name: params.businessName,
      business_type: params.businessType,
      // honeypot — must remain empty
      website: '',
    });
    const d = res.data;
    return {
      user: mapUser(d.user),
      token: d.token,
      tenant: d.tenant ? mapTenant(d.tenant) : null,
      emailVerified: Boolean(d.user?.email_verified),
    };
  }

  async verifyEmail(email: string, code: string): Promise<LoginResult> {
    const { data: res } = await api.post('/auth/verify-email', { email, code });
    const d = res.data;
    return {
      user: mapUser(d.user),
      token: d.token,
      tenant: d.tenant ? mapTenant(d.tenant) : null,
    };
  }

  async resendVerification(email: string): Promise<void> {
    await api.post('/auth/verify-email/resend', { email });
  }

  async requestPasswordReset(email: string): Promise<void> {
    await api.post('/auth/password/forgot', { email });
  }

  async resetPassword(input: { email: string; token: string; password: string }): Promise<void> {
    await api.post('/auth/password/reset', input);
  }

  async logout(): Promise<void> {
    await api.post('/auth/logout');
  }

  async me(): Promise<{ user: import('@/domain/entities/user').User; tenant: import('@/domain/entities/tenant').Tenant | null }> {
    const { data: res } = await api.get('/auth/me');
    const d = res.data;
    return {
      user: mapUser(d.user ?? d),
      tenant: d.tenant ? mapTenant(d.tenant) : null,
    };
  }
}

File: apps/admin_mobile/lib/infrastructure/api/repositories/api_auth_repository.dart
import '../../../domain/entities/user.dart';
import '../../../domain/entities/tenant.dart';
import '../../../domain/repositories/auth_repository.dart';
import '../dio_client.dart';
import '../mappers/user_mapper.dart';
import '../mappers/tenant_mapper.dart';

class ApiAuthRepository implements AuthRepository {
  final DioClient _client;

  ApiAuthRepository(this._client);

  @override
  Future<LoginResult> login(String email, String password) async {
    final response = await _client.dio.post('/auth/login', data: {
      'email': email,
      'password': password,
    });
    final data = response.data['data'] ?? response.data;
    return LoginResult(
      user: mapUser(data['user'] as Map<String, dynamic>),
      token: data['token'] as String,
      tenant: data['tenant'] != null
          ? mapTenant(data['tenant'] as Map<String, dynamic>)
          : null,
    );
  }

  @override
  Future<LoginResult> register({
    required String name,
    required String email,
    required String password,
  }) async {
    final response = await _client.dio.post('/auth/register', data: {
      'name': name,
      'email': email,
      'password': password,
    });
    final data = response.data['data'] ?? response.data;
    return LoginResult(
      user: mapUser(data['user'] as Map<String, dynamic>),
      token: data['token'] as String,
      tenant: data['tenant'] != null
          ? mapTenant(data['tenant'] as Map<String, dynamic>)
          : null,
    );
  }

  @override
  Future<void> logout() async {
    await _client.dio.post('/auth/logout');
  }

  @override
  Future<({User user, Tenant? tenant})> me() async {
    final response = await _client.dio.get('/auth/me');
    final data = response.data['data'] ?? response.data;
    return (
      user: mapUser(data['user'] as Map<String, dynamic>),
      tenant: data['tenant'] != null
          ? mapTenant(data['tenant'] as Map<String, dynamic>)
          : null,
    );
  }
}
